path-utils.js

```diff
@@ -1,227 +1,242 @@
-			"userinfo",
-			"username",
-			"password",
-			"host",
-			"hostname",
-			"port",
-			"pathname",
-			"directory",
-			"filename",
-			"filenameExtension",
-			"search",
-			"hash"
-	],
-
-	defaultPorts: { http: "80", https: "443", ftp: "20", ftps: "990" },
-
-	parseUrl: function( url ) {
-
-		// If we're passed an object, we'll assume that it is
-		// a parsed url object and just return it back to the caller.
-		if ( url && typeof url === "object" ) {
-			return url;
-		}
-
-		var matches = PathUtils.urlParseRE.exec( url || "" ) || [],
-			props = PathUtils.parsedUrlPropNames,
-			cnt = props.length,
-			result = {},
-			i;
-	
-		for ( i = 0; i < cnt; i++ ) {
-			// Most browsers returns an empty string for empty sub-matches, but
-			// IE returns undefined, so we need to make sure we normalize empty
-			// sub-matches so results are consistent across all browsers.
+/** 
+ * @license
+ * path-utils.js - version 0.0.2
+ * Copyright (c) 2011, Kin Blas
+ * All rights reserved.
+ * 
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the <organization> nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+(function (root, factory) {
+	if (typeof define === 'function' && define.amd) {
+		// AMD. Register as an anonymous module.
+		define([], factory);
+	} else if (typeof module === 'object' && module.exports) {
+		// Node. Does not work with strict CommonJS, but
+		// only CommonJS-like environments that support module.exports,
+		// like Node.
+		module.exports = factory();
+    } else {
+		// Browser globals (root is window)
+		root.returnExports = factory();
```

Comment:
@jblas oh damn, my bad. I must have nuked over the last change after doing a ´git reset` when I forgot to minimize the changes I made: the `root.ReturnExports` line is supposed to be `root.PathUtils`, which exposes `PathUtils` to window (`root` is `this` in browser, which and in browser `window === this`)

This has now been fixed in 6f466d6 (force push)